WordPress/gutenberg
# Patterns: Allow for filtering of block patterns by source

Related:
- https://github.com/WordPress/gutenberg/pull/51144
- https://github.com/WordPress/gutenberg/pull/51078

Backport PR for this change: https://github.com/WordPress/wordpress-develop/pull/4697

## What?

- Extends the REST API endpoint for block patterns to provide a new `source` attribute
- When registering core block patterns, the new `source` attribute 

## Why?

The Site Editor Library has been designed to allow users a more refined experience in managing their patterns. This requires the ability to filter out core block patterns which will not be editable by users.

## How?

- Adds a new source attribute to the returned items from the pattern endpoint
- Sets the source attribute when loading core patterns
- This attribute will be used to filter patterns in https://github.com/WordPress/gutenberg/pull/51078


## Testing Instructions

1. Test via https://github.com/WordPress/gutenberg/pull/51078
2. Open the Site Editor and navigate to the Library
3. Explore the library categories and ensure that only template parts, theme patterns, and reusable blocks are shown.
